File: DSA/week4.2/first_unique_char.js
// leetcode problem - 387
var firstUniqChar = function(s) {
    const obj = {}
    for(let char in s){

        if(obj[s[char]]){
            obj[s[char]]+=1
        }else{
            obj[s[char]]=1
        }
    }
    for(let i=0;i<s.length;i++){
        if(obj[s[i]]===1){
            return i
        }
    }
    return -1
};
console.log(firstUniqChar("leetcode"))

File: DSA/week4.2/first_unique_char.py
# leetcode problem - 387
class Solution(object):
    def firstUniqChar(self, s:str)->int:
        """
        :type s: str
        :rtype: int
        """
        hashmap = {}
        for char in s:
            if char in hashmap:
                hashmap[char] += 1
            else:
                hashmap[char] = 1
        for i in range(len(s)):
            if hashmap[s[i]] == 1:
                return i
        return -1

s= Solution()
print(s.firstUniqChar("leetcode"))